app/models/document.rb

```diff
@@ -100,6 +100,15 @@ def published?
     )
   end
 
+  # this is to support linking to documents which haven't get been published,
+  # but will be published within 5 seconds
+  def published_very_soon?
+    Edition
+      .scheduled
+      .where("scheduled_publication <= ?", 5.seconds.from_now)
```

Comment:
I'd have probably gone for something like 30 seconds, but 5 is fine as well :+1: